core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1764,17 +1764,21 @@ class SparkContext(config: SparkConf) extends Logging {
     listenerBus.post(SparkListenerUnpersistRDD(rddId))
   }
 
+  def addJar(path: String) {
+    addJar(path, false)
+  }
+
   /**
    * Adds a JAR dependency for all tasks to be executed on this `SparkContext` in the future.
    *
    * If a jar is added during execution, it will not be available until the next TaskSet starts.
    *
    * @param path can be either a local file, a file in HDFS (or other Hadoop-supported filesystems),
    * an HTTP, HTTPS or FTP URI, or local:/path for a file on every worker node.
-   *
+   * @param check when true, check jar file exists
    * @note A path can be added only once. Subsequent additions of the same path are ignored.
    */
-  def addJar(path: String) {
+  def addJar(path: String, check: Boolean) {
```

Comment:
I'm not sure if it is good to add another parameter for `check` only, or we just change the semantics of `addJar`?

Reply:
@jerryshao That is why I add another function to cover old addJar Function.

Reply:
@jerryshao All change done,  and I think origin Unit Test is ok. 
`  def addJar(path: String) {
    addJar(path, false)
  }
`

These method just replace origin addJar func.

Reply:
No, you should add more test to cover `check=true` scenario.

Reply:
@jerryshao  Got it

Reply:
@jerryshao Sould I add more ?